### PR TITLE
Make plugin installation tests less fragile

### PR DESCRIPTION
**Pull request by basil**
### Context

Prerequisite to #7477. While working on #7477 I tripped over these fragile tests yet again, so I decided to make them less fragile in preparation for that PR. These changes can be rationalized independently but their underlying motivation is to ease the implementation of #7477.

### Problems

While implementing #7477 I found it necessary to update the `credentials` plugin on the test classpath from the very ancient 1.9.4 to the latest version for reasons that I will explain in that pull request. This would have been straightforward if this was a simple `test`-scoped dependency, but it was not because the old version was checked in as `test/src/test/resources/plugins/credentials.hpi` and used in various strange ways.

Failures I got when updating this test dependency and normalizing it to a regular `test`-scoped Maven dependency:

- `PluginManagerTest#uploadDependencyResolution` started failing because it relied on being able to exclude `test`-scoped Maven dependencies and upload a set of interdependent plugins all out of `test/src/resources/plugins/*.hpi`, which stopped working once `credentials.hpi` was deleted from the source tree.
- `PluginManagerTest#testDeprecationNotices` similarly relied on being able to exclude `test`-scoped Maven dependencies and install a set of plugins all out of `test/src/resources/plugins/*.hpi`, which stopped working once `credentials.hpi` was deleted from the source tree.
- `Security914Test` was trying to install `credentials.hpi` out of `test/src/resources/plugins/`.

### Solutions

- Of these the easiest to fix was `Security914Test`: with `credentials` now a `test`-scoped Maven dependency, the `@WithPlugin("credentials.hpi")` line could simply be deleted. The test than had to be adjusted to work with the newer version of `credentials` which uses SVGs rather than PNGs.
- `PluginManagerTest#uploadDependencyResolution` could be trivially made to work again by adapting it to use the fake `mandatory-depender` and `dependee` plugins rather than ancient versions of real plugins, once an update was made to its corresponding `upload-test-update-center.json` resource.
- `PluginManagerTest#testDeprecationNotices` was harder to fix in place but I realized that the main logic it was testing was actually the logic to parse `deprecations-update-center.json` rather than anything in` PluginManager` itself (which just delegates to `UpdateSite` to determine deprecation). So we can preserve the same level of code coverage but make the test a lot simpler by moving it to `UpdateSiteTest`. I did that here, keeping the same JSON file as input.

The result of all of this is that, with this change applied, the Guice upgrade trivially passes. So I plan to first integrate this PR, then rebase the Guice PR on top of this.

### Testing done

I successfully tested this PR both by itself and in concert with the Guice upgrade, with a successful core CI build and full BOM/PCT matrix passing.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7483"></a>

### CODE REVIEW

**Comment by basil** [comment]
Thank you for the review Stefan. Yes the call to `getInfo()` was incidental to the primary purpose of the test and I did not deem it worth keeping. From my perspective there is no change that needs to be made to this PR.

**Review by StefanSpieker — APPROVED** [review]
Ok, I also like that the new test now also runs on Windows!

File: test/src/test/java/hudson/model/UpdateSiteTest.java
Comment (daniel-beck) [inline test/src/test/java/hudson/model/UpdateSiteTest.java:225]: This loses test coverage for plugins not currently being published, but marked as deprecated via top-level metadata.

While we can assume that's kinda taken care of by the `icon-shim` test having the top-level deprecation URL, an explicit assertion would be nicer. Given `mandatory-depender`, `dependee`, etc. plugins are around, could they stand in for real plugins for this purpose?

FWIW while it's mostly true that

> the call to `getInfo()` was incidental to the primary purpose of the test and I did not deem it worth keeping.

it confirmed absence of the plugin from update site metadata.

File: test/src/test/java/hudson/PluginManagerTest.java
Comment (daniel-beck) [inline test/src/test/java/hudson/PluginManagerTest.java:602]: Is this not useful to see at a glance what cases we're testing?

**Review by daniel-beck — APPROVED** [review]
I'd prefer the case of suspended distribution remains tested. Overall fine though.

File: test/src/test/java/hudson/model/UpdateSiteTest.java
Comment (basil) [reply test/src/test/java/hudson/model/UpdateSiteTest.java]: I added an assertion for a plugin not currently being published, but marked as deprecated via top-level metadata.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by daniel-beck — APPROVED** [review]

**Pull request merged**
